pkg/api/validation/validation.go

```diff
@@ -1278,6 +1291,18 @@ func ValidateSecret(secret *api.Secret) errs.ValidationErrorList {
 		}
 	case api.SecretTypeOpaque, "":
 		// no-op
+	case api.SecretTypeDockercfg:
+		dockercfgBytes, exists := secret.Data[api.DockerConfigKey]
+		if !exists {
+			allErrs = append(allErrs, errs.NewFieldRequired(fmt.Sprintf("data[%s]", api.DockerConfigKey)))
+			break
+		}
+
+		dockercfg := &credentialprovider.DockerConfig{}
+		if err := json.Unmarshal(dockercfgBytes, dockercfg); err != nil {
```

Comment:
It'd be one thing if we validated it here and that made it so that kubelet didn't have to deal with the error cases. But kubelet has to deal with them anyway, so it doesn't gain much here to make this check. Plus see my earlier point: this step as written does not make any assertions about the contents of the json, it only verifies that it parses. You can literally stick ANY valid json in here and it will pass. I'm not saying there's zero value in that, but we can get the same benefit with map[string]interface{} without worrying about making this struct part of the API.